typings/index.d.ts

```diff
@@ -1708,7 +1708,9 @@ declare module 'discord.js' {
 
 	export class GuildChannelStore extends DataStore<Snowflake, GuildChannel, typeof GuildChannel, GuildChannelResolvable> {
 		constructor(guild: Guild, iterable?: Iterable<any>);
-		public create(name: string, options?: GuildCreateChannelOptions): Promise<TextChannel | VoiceChannel | CategoryChannel>;
+		public create(name: string, options?: GuildCreateChannelOptions & { type: 'voice' }): Promise<VoiceChannel>;
+		public create(name: string, options?: GuildCreateChannelOptions & { type: 'category' }): Promise<CategoryChannel>;
+		public create(name: string, options?: GuildCreateChannelOptions): Promise<TextChannel>;
```

Comment:
As example this snippet:
```ts
const opts: GuildCreateChannelOptions = { type: 'category' }
const channel = await guild.channels.create('test', opts); // channel being TextChannel (not a CategoryChannel or an union of possible types)
```

It's not possible to infer what channel is being created here.
This should probably still return an union of all available types.

To solve this:
- Should this overload return `Promise<TextChannel | VoiceChannel | CategoryChannel>`
  > `guild.channels.create('test', opts);`
- And another one being added similar to the above ones returning a `Promise<TextChannel>`,
  However the `options` actually need to be optional in that case. See the first example below
  > `guild.channels.create('test');` or `guild.channels.create('test', { type: 'text' });`